src/Corax/Queries/SortingMatch/SortingMatch.cs

```diff
@@ -287,12 +299,15 @@ private struct Fetcher : IFetcher
             items = new Span<MatchComparer<TComparer, TOut>.Item>(match._buffer, match._bufferUsedCount);
 
             int matchesToReturn = Math.Min(match._bufferUsedCount, matches.Length);
+            
+            ref var itemsStart = ref MemoryMarshal.GetReference(items);
             for (int i = 0; i < matchesToReturn; i++)
             {
                 // We are getting the top element because this is a heap and the top element is the one that is
                 // going to be removed. Then the rest of the element will be pushed up to find the new top.
-                matches[i] = items[0].Key;
-                HeapDown(items, match._bufferUsedCount - i - 1);
+                ref var matchIdxPtr = ref Unsafe.Add(ref matchesRef, i);
+                matchIdxPtr = items[0].Key;
```

Comment:
itemsStart  here, no?

Reply:
Yeah, we could use that. But in this case the check is done outside of the loop (loop hoisting). So no need to uglify that much.